### DISCUSSION TITLE
Building using Docker Image

### DISCUSSION THREAD

**Discussion by jones139** [InfiniTimeOrg/InfiniTime#1956] [General]
At the moment I am building Infinitime using the docker file that you provide (thank you - this saved me working out what was wrong with my local build environment!).

It seems to build ok and I can flash it ok with GadgetBridge on Android, but I am seeing strange behaviour if I modify the code, increment the version number in CMakeLists.txt and re-run the build.

What I see is it builds a pinetime-mcuboot-dfu-x.y.z.zip file as you would expect for the new version, but when I flash it onto the device it appears to be the old version (the about screen shows the old version number and build date/time).
If I delete the build folder and re-run the build from scratch, it works as expected (but of course takes quite a lot longer to run because it has to compile everything).

To start with I thought I had made a mistake and had not copied the correct file onto my phone to flash using GadgetBridge, but it seems to be quite repeatable - I see issues if I re-run build on top of a previous build, but it works as expected if I run it from scratch.

Am I doing something wrong?

Thanks,  Graham.

**Comment by JF002** [comment]
I don't think you're doing anything wrong, since I've already observed this behavior. I think this is cause by the way the file [version.h](https://github.com/InfiniTimeOrg/InfiniTime/blob/main/src/Version.h.in) is generated by CMake. It looks like CMake does not re-generate it when the version of the project was changed. So InfiniTime builds the new code with an un-changed Version.h file and thus still displays the old version.

There must be a way to fix this in CMake, but I haven't tried to figure that out yet.

**Reply by jones139** [reply]
I have started to have a look.   I think that Version.h is being re-generated correctly, but it appears that Version.h is not a make dependency for SystemInfo.cpp.o
I'll see if I can work out how dependencies work in CMake and propose a fix...

**Reply by jones139** [reply]
Pull Request #1960 should fix this issue (I think - it seems to be working for me anyway).
